### ISSUE TITLE
Podfile has incorrect path to node_modules with `init-test-app`

### ISSUE THREAD

**Issue by Saadnajmi** [microsoft/react-native-test-app#369]
### Description

I was trying to integrate RNTA locally on a community module (this one https://github.com/react-native-menu/menu) with the steps listed in the repo. I followed the steps listed in the repo README, and tried to do `pod install` in the iOS subdirectory of the new test app. That failed because it thought the node_module for RNTA was one level up, instead of two levels up.

I.E:
```
require_relative '../node_modules/react-native-test-app/test_app'
# Instead of
require_relative '../../node_modules/react-native-test-app/test_app'
```
The steps have you add RNTA as a dependency to the root of the repo, then generates a sub package for the test app when you run `init-test-app`. That new sub package does not have RNTA as a dependency, so the `require_relative` fails.

### Your environment

* `react-native-test-app` version: 0.6.0
* Tested platforms:
  * [ ] Android
  * [x] iOS
  * [ ] macOS
  * [ ] Windows
* System information:
  
 ```
System:
    OS: macOS 11.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 447.41 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.4.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.18.1 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.3, 30.0.1
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1.2 4.1.2
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.0 => 0.64.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
### Steps to reproduce

```
git clone git@github.com:react-native-menu/menu.git
cd menu
mv example/ example2/ # This is to just move the existing test app out of the way temporarily
yarn add react-native-test-app --dev
yarn init-test-app
# The options I chose
# ✔ What is the name of your test app? … MenuExample
# ✔ Which platforms do you need test apps for? › Android, iOS, macOS
# ✔ Where should we create the new project? … example
yarn
cd ios
pod install
```
I get this error:
```
saads-macbook-pro:ios sanajmi$ pod install

[!] Invalid `Podfile` file: cannot load such file -- /Users/sanajmi/workspace/menu/example/node_modules/react-native-test-app/test_app.

 #  from /Users/sanajmi/workspace/menu/example/ios/Podfile:1
 #  -------------------------------------------
 >  require_relative '../node_modules/react-native-test-app/test_app'
 #  
 #  -------------------------------------------
saads-macbook-pro:ios sanajmi$ 
```

Adding ../ to the path fixes this.

**Comment by tido64** [comment]
Is `@react-native-menu/menu` set up as a monorepo? Otherwise, did you run `yarn` inside `/example`?

**Comment by Saadnajmi** [comment]
I don't think it's a monorepo, it's just the one component.
I did run `yarn` inside `example`, but RNTA is not listed as a dependency in `example/package.json`

**Comment by tido64** [comment]
I can't repro it…
```
react-native-menu % yarn init-test-app
yarn run v1.22.10
$ /~/react-native-menu/node_modules/.bin/init-test-app
✔ What is the name of your test app? … MenuExample
✔ Which platforms do you need test apps for? › Android, iOS, macOS
✔ Where should we create the new project? … example
✨  Done in 13.03s.
react-native-menu % cd example
example % grep react-native-test-app package.json
    "react-native-test-app": "^0.6.0",
example % yarn
...
example % cd ios
ios % pod install
...
Pod installation complete! There are 53 dependencies from the Podfile and 44 total pods installed.
ios %
```

**Comment by Saadnajmi** [comment]
Interesting. Perhaps it's because I had that failure earlier in https://github.com/microsoft/react-native-test-app/issues/367 that `init-test-app` didn't get as far as adding the dep to my package.json? I can still repro it locally, but I don't successfully complete `init-test-app`

**Comment by tido64** [comment]
That could be. I manually copied in the missing files since #368 hasn't landed yet.

**Comment by tido64** [comment]
Can you try again with [latest](https://www.npmjs.com/package/react-native-test-app/v/0.6.2)?

**Comment by Saadnajmi** [comment]
yep, looks like it was fixed with #368!
